src/Data/Group.hs

```diff
@@ -339,6 +339,9 @@ instance (Group a, Group b, Group c, Group d, Group e) => Group (a,b,c,d,e) wher
   {-# inline invert #-}
 
 #if MIN_VERSION_base(4,12,0)
+instance (Applicative f, Group a) => Group (Ap f a) where
+  invert (Ap fa) = Ap (invert <$> fa)
```

Comment:
Question: 

Is `f a` where `a` is `mempty` also considered `empty` for the `Alternative` instance? If this is the case, then that would imply that 

```haskell
f a <> (invert <$> f a)
~ liftA2 (\a a -> a <> invert a) fa fa
~ f mempty
~ empty
```

So that `Ap f` has a `Cancellative` instance.

Reply:
Actually, @taneb noted this is not lawful in general for any old `Applicative`: https://github.com/Taneb/groups/pull/5#issuecomment-739718470

In fact, this may need to be `Cancellative` to ensure inversion is sound.